neurodamus/metype.py

```diff
@@ -35,6 +36,60 @@ def connect2target(self, target_pp=None):
         """Connects empty cell to target"""
         return Nd.NetCon(self._cellref, target_pp)
 
+    @staticmethod
+    def get_sec(cell, section_id):
+        """Return the section corresponding to the section_id."""
+        sec_lists = [
+            getattr(cell, stype) for stype in BaseCell._section_types if hasattr(cell, stype)
+        ]
+        total_len = sum(len(lst) for lst in sec_lists)
+
+        if not (0 <= section_id < total_len):
+            raise IndexError(
+                f"Global index {section_id} out of range (max index is {total_len - 1})"
+            )
+
+        offset = 0
+        for sec_list in sec_lists:
+            if section_id < offset + len(sec_list):
+                return sec_list[section_id - offset]
+            offset += len(sec_list)
```

Comment:
In the emodel hoc template, there are variables like `nSecSoma`, `nSecAxon`, etc. for the counts of different section types. Could we use them to determine the type of section and the local section id instead of creating a full `sec_lists`?
 
Another question, for `axon`s, is the global axon `section_id` corresponding to the original axon list or after `replace_axon` actions in the emodel template?

Reply:
this is the `get_sec` function just below no? Otherwise I do not get the question. 

Also, yes, now the sequence is fixed in `_section_layout `

Reply:
Yes, this new version of `get_sec` looks good. Thanks!